web-ui/src/main/resources/catalog/js/edit/EditorController.js

```diff
@@ -167,6 +167,15 @@
             }).then(function(data) {
               $scope.metadataFound = data.count !== '0';
               $scope.metadataNotFoundId = $routeParams.id;
+              
+              $http({
+                method: 'GET',
+                url: 'metadata/' + $routeParams.id + '/lock'
+              }).then(function successCallback(response) {
+                  $scope.metadataLocked = response.data;
```

Comment:
You should stop opening the editor if locked ? There is no need to continue and try to open editor which return error 500. 

Would it make sense to add the lock status in the search response in order to:
* warn user that a record is locked (even before clicking the edit button)
* will avoid making 3 requests before opening the editor

I think it was part of the initial work no (http://trac.osgeo.org/geonetwork/wiki/ImprovedWorkflow) with icons if records is under edits.